src/main/java/org/cryptomator/cryptofs/health/dirid/OrphanDir.java

```diff
@@ -134,10 +158,53 @@ Path prepareRecoveryDir(Path pathToVault, FileNameCryptor cryptor) throws IOExce
 		return new CryptoPathMapper.CiphertextDirectory(stepParentUUID, stepParentDir);
 	}
 
+	//visible for testing
+	Optional<String> retrieveDirId(Path orphanedDir, Cryptor cryptor) {
+		var dirIdFile = orphanedDir.resolve(Constants.DIR_ID_FILE);
+		var dirIdBuffer = ByteBuffer.allocate(36); //a dir id contains at most 36 ascii chars, in this impl encoded in utf8
+
+		try (var channel = Files.newByteChannel(dirIdFile, StandardOpenOption.READ); //
+			 var decryptingChannel = createDecryptingReadableByteChannel(channel, cryptor)) {
+			ByteBuffers.fill(decryptingChannel, dirIdBuffer);
+			dirIdBuffer.flip();
+		} catch (IOException e) {
+			LOG.info("Unable to read dirIdFile of {}.", orphanedDir, e);
+			return Optional.empty();
+		}
+
+		var allegedDirId = StandardCharsets.UTF_8.decode(dirIdBuffer).toString();
```

Comment:
when you consider anything longer than 36 bytes incorrect, you shouldn't allow UTF-8 but rather use ASCII, since 36 UTF-8 chars may be 144 bytes (or even more).

Reply:
Fixed in [85745fb](https://github.com/cryptomator/cryptofs/pull/127/commits/85745fb1b6c7aef559e9744924ac19879f50db5e)